rules/alerting.go

```diff
@@ -276,6 +298,7 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 		}
 
 		vec = append(vec, r.sample(a, ts))
+		vec = append(vec, r.forStateSample(a, ts, float64(a.ActiveAt.Unix())))
```

Comment:
As we'll be bumping this forwards at restart, the intended firing time might be easier to understand. It's weird for ActiveAt to change

Reply:
If we want to change the intended firing time, it would require changing `holdDuration`, which wont be right. I can't think of any other way to achieve this without adding another variable in the `AlertingRule` for an addition expected firing time.